vissl/hooks/state_update_hooks.py

```diff
@@ -95,6 +95,11 @@ def on_phase_start(self, task: "tasks.ClassyTask") -> None:
             if hasattr(task.dataloaders[phase_type].sampler, "set_epoch"):
                 # task.phase_idx is current running phase id
                 task.dataloaders[phase_type].sampler.set_epoch(task.phase_idx)
+        if hasattr(task.dataloaders[phase_type], "dataset"):
+            if hasattr(task.dataloaders[phase_type].dataset, "set_epoch"):
```

Comment:
how does this work with non-airstore datasets? will the conditional be still satisfied since dataset is "GenericSSLDataset" which has the `set_epoch` ?

Reply:
Yes, GenericSSLDataset will go through each sub dataset objs, and if it implemented "set_epoch" it will dispatch down

Reply:
I change the code directly check dataset types